### ISSUE TITLE
no-unregistered-classes with detectComponentClasses and selectors

### ISSUE THREAD

**Issue by aaronmhargrove** [schoero/eslint-plugin-better-tailwindcss#241]
Hello, I'm in the process of adding this plugin, and I think I've come across a minor issue. When using `detectComponentClasses` it does successfully find my custom classes in the globals css file. However, it's flagging instances of those custom classes that are used with a selector.

```css
/* globals.css */

@layer components {
  .example-class {
    /* ..... */
  }
}
```

This works: 
```tsx
// Correctly allows the class ✅
<MyComponent className="example-class" />
```

This get's reported as a violation:
```tsx
// Flags the class as unregistered ❌
<MyComponent className="md:example-class" />
```
